staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -122,11 +113,6 @@ import (
 // to something else, for example through a
 // `client-go/util/workqueue`.
 //
-// Each query to an informer's local cache --- whether a single-object
-// lookup, a list operation, or a use of one of its indices --- is
-// answered entirely from one of the collection states received by
-// that informer.
-//
```

Comment:
This is still useful, even if "collection states" did not necessarily exist in the API servers.
It means that while the results of an informer lister's `.List()` or `.Get()` are populated the informer's cache from which those results are populated is not modified. This guarantees, for instance, that `List()` results will never include two API objects with the same namespaced name.

Reply:
Is there any implication other than "`List()` results will never include two API objects with the same namespaced name" [and APIGroup and Kind] ?

Reply:
I haven't thought about this long enough to be confident that there aren't other implications, but another one that comes to my mind is that `List()` is guaranteed to return.

Reply:
I have added a paragraph above that attempts to say what can be said in this regard.